### PR TITLE
[automate-3066] Delete authz v1 server code

### PR DESCRIPTION
**Pull request by tylercloke**
### :nut_and_bolt: Description: What code changed, and why?

Remove v1 authz code and port any still relevant tests.

### :white_check_mark: Checklist

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [ ] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### Passing Builds

https://buildkite.com/chef-oss/chef-automate-master-verify/builds/10185
https://buildkite.com/chef/chef-automate-master-verify-private/builds/10533

### CODE REVIEW

File: components/event-gateway/pkg/nats/authenticator.go [hunk 1]
```diff
@@ -292,36 +277,17 @@ func (a *automateAuthenticator) tryAuthzV2(ctx context.Context, subjects []strin
 	})
 	if err != nil {
 		if status.Convert(err).Code() == codes.FailedPrecondition {
-			return err, true
+			return err
 		}
-		return errors.Wrapf(err, "authorizing action %q on resource %q for members %q", actionV2, resourceV2, subjects), false
+		return errors.Wrapf(err, "authorizing action %q on resource %q for members %q", actionV2, resourceV2, subjects)
```
Comment (msorens) [inline components/event-gateway/pkg/nats/authenticator.go]:
```suggestion
		return errors.Wrapf(err, "authorizing action %q on resource %q for members %q", action, resource, subjects)
```

File: components/authz-service/engine/opa/opa.go [hunk 2]
```diff
@@ -127,26 +127,26 @@ func (s *State) initModules() error {
...
-		rego.Store(s.store),
-	)
-	pr, err := r.PartialResult(ctx)
-	if err != nil {
-		return errors.Wrap(err, "partial eval")
-	}
-	s.partialAuth = pr
-	return nil
-}
+// SetPolicies(), we want to do as little work per call as possible.
```
Comment (blakestier) [inline components/authz-service/engine/opa/opa.go]: Why is this commented out instead of deleted?

Comment (tylercloke) [reply components/authz-service/engine/opa/opa.go]: Should be deleted. Forgot to. Thanks!

File: components/authz-service/engine/engine.go [hunk 3]
```diff
@@ -4,43 +4,15 @@ import (
...
-// interfaces, and can thus be plugged into either of those constructors.
+// Engine abstracts different v2 decision engines.
 type Engine interface {
-	V1Engine
-
 	// V2Authorizer and V2Writer are never used together (the authz section of the
 	// service needs V2Authorizer, the policy section cares about V2Writer), so we
 	// collect them here instead of introducing a V2Engine interface.
 	V2Authorizer
 	V2p1Writer
```
[the remark this answers is not in the document]
Comment (tylercloke) [reply components/authz-service/engine/engine.go]: I'm trying to minimize rebase issues. We should definitely update those at some point. Maybe now is as good of time as any 🤷‍♂ .

File: components/authz-service/engine/opa/opa.go [hunk 4]
```diff
@@ -127,26 +127,26 @@ func (s *State) initModules() error {
...
+// func (s *State) initPartialResult(ctx context.Context) error {
```
[the remark this answers is not in the document]
Comment (tylercloke) [reply components/authz-service/engine/opa/opa.go]: Yup

File: components/authz-service/engine/engine.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (tylercloke) [reply components/authz-service/engine/engine.go]: Updated

File: api/interservice/authz/v2/authz.proto [hunk 5]
```diff
@@ -1,10 +1,26 @@
 syntax = "proto3";
 
 import "validate/validate.proto";
+import "api/external/common/version/version.proto";
 
 package chef.automate.domain.authz.v2;
 option go_package = "github.com/chef/automate/api/interservice/authz/v2";
 
+// TODO (tc): Keeping this around for the validation used in protoc-gen-policy/policy.go
+message IsAuthorizedReq {
```
[the remark this answers is not in the document]
Comment (tylercloke) [reply api/interservice/authz/v2/authz.proto]: How can I make the comment more helpful there? It's still being used here:

https://github.com/chef/automate/blob/master/components/automate-grpc/protoc-gen-policy/policy/policy.go#L68-L75

**Pull request merged**